lib/modules/manager/cargo/artifacts.ts

```diff
@@ -130,13 +131,23 @@ async function updateArtifactsImpl(
         config.constraints?.rust,
       );
     } else {
-      const missingDep = updatedDeps.find((dep) => !dep.lockedVersion);
-      if (missingDep) {
-        // If there is a dependency without a locked version then log a warning
-        // and perform a regular workspace lockfile update.
-        logger.warn(
-          `Missing locked version for dependency \`${missingDep.depName}\``,
-        );
+      const nonCrateDep = updatedDeps.find(
+        (dep) => dep.datasource !== CrateDatasource.id,
+      );
+      const crateDepWithoutLockedVersion = updatedDeps.find(
+        (dep) => !dep.lockedVersion && dep.datasource === CrateDatasource.id,
+      );
+      // Non-crate dependencies (like git ones) do not have locked versions.
+      // For crate dependencies, not having a locked version is not expected.
+      // In both situations, perform a regular workspace lockfile update.
+      if (nonCrateDep || crateDepWithoutLockedVersion) {
```

Comment:
Git dependencies do not have locked versions in the lock file, so without this, while we would correctly call `cargoUpdate` instead of `cargoUpdatePrecise`, without this change, we would display a warning that appears in the dependency dashboard when updating git dependencies.

Not really a huge fan of the overall logic, but I did not really find cleaner ways to achieve this.

Reply:
Could you split this back to Discussion - either existing or new - assuming that a design decision needs making? It's hard to follow threads in PR reviews